packages/back-end/src/models/FeatureRevisionModel.ts

```diff
@@ -234,28 +236,44 @@ export async function createRevision({
       rules,
     }),
   };
+  baseVersion = baseVersion || feature.version;
+  const baseRevision = (await getRevision(
+    feature.organization,
+    feature.id,
+    baseVersion
+  )) as FeatureRevisionInterface;
   let status = "draft";
-  if (publish && !requiresReview) {
-    status = "published";
-  } else if (publish && requiresReview) {
-    status = "pending-review";
-  }
-  const doc = await FeatureRevisionModel.create({
+  const revision = {
     organization: feature.organization,
     featureId: feature.id,
     version: newVersion,
     dateCreated: new Date(),
     dateUpdated: new Date(),
-    datePublished: status === "published" ? new Date() : null,
+    datePublished: null,
     createdBy: user,
     baseVersion: baseVersion || feature.version,
     status,
-    publishedBy: status === "published" ? user : null,
+    publishedBy: null,
     comment: comment || "",
     defaultValue,
     rules,
     log: [log],
+  } as FeatureRevisionInterface;
+  const requiresReview = checkIfRevisionNeedsReview({
+    feature,
+    baseRevision,
+    revision,
+    settings: org.settings,
   });
+  if (publish && !requiresReview) {
+    status = "published";
+    revision.publishedBy = user;
+    revision.datePublished = new Date();
+  } else if (publish && requiresReview) {
+    status = "pending-review";
```

Comment:
Isn't this just updating a local variable?  Don't you want to update `revision.status` instead? Same above